### DISCUSSION TITLE
G7 needs permanent location service?

### DISCUSSION THREAD

**Discussion by kameamea** [NightscoutFoundation/xDrip#2986] [Q&A]
With the recommended settings and almost the latest nightly I lose connection as soon as I disable location service. Since other apps drain a lot of battery out of the location service I used to switch it on only for pairing a new sensor, which used to work well with G6 and G7.
I think the behavior changed when i started using the recommended settings. But i don't know which of the recommended settings causes the need for the location service. Any ideas or different experiences?

Phone: Huawei P30
Xdrip 20230716

**Comment by Navid200** [comment]
Android devices need the location service to be enabled to use Bluetooth as far as I know I believe up to some earlier versions of Android.
I have no idea if this requirement has gone away with Android 13.
I thought you needed it even if you wanted to use the Dexcom app.

**Reply by kameamea** [reply]
I'm not on Android 13. The Dexcom app needs the location service only while pairing, afterwards it can be turned off. Bluetooth headphones also work without location service. And I'm pretty sure xdrip used to work well without GPS, apart from the pairing phase.

**Reply by Navid200** [reply]
There is a big difference between the location service and GPS.
xDrip has never needed a GPS app and doesn't need it now either.

As far as I can tell you, we are not going to remove the requirement for Android location service.

If you want to use xDrip, I suggest that you use our recommended settings.
If xDrip doesn't fit your needs, you can use alternatives.

I have been using xDrip for over 4 years now, 24/7.  I have all this time had the location service enabled.
I have done this with 6 or 7 different phones going from Android 5 to Android 13 now.
I charge my phone every other day.
There is no power waste caused by xDrip..

**Comment by jamorham** [comment]
xDrip needs the permission for location due to this becoming a poorly thought out requirement imposed on us by Android 6 to use Bluetooth. Google have since rethought this for Android 12+ but we still support the legacy mechanism for compatibility reasons. xDrip doesn't activate the location service for the bluetooth activity and the only time location is used is with the emergency messages feature which only activates if you have enabled it. I have never detected any battery drain related to xDrip's location permission so I don't see a reason to change this from how it works at the moment.

**Reply by kameamea** [reply]
I agree that xdrip doesn't drain battery due to location service. But for other apps it can make a huge difference, for example navigation software osmand but also the Dexcom g7 software. That's why i prefer to activate location service only when i really need it. Besides there is some privacy concern.
And i know we need location service for Bluetooth pairing. But once paired Bluetooth applications still can work when location service is turned off. And I'm pretty sure xdrip behaved this way with Dexcom g6 (and probably g7).
So is there some active check for location service in xdrip which stops the connection (this may not be the proper term but i mean it delivers no more new values) and does this depend on some settings?

**Reply by Mocadon** [reply]
Well, it's 2025 and I'm running GrapheneOS Based on hardened Android 15 AOSP and xDrip+ because of privacy concerns. So, the stable release I'm running was September 2024. Can we turn off the location permission yet? If not, I request that be removed as a requirement for Android 12+. It's not the highest priority, just something that probably should happen IMHO.

**Reply by Navid200** [reply]
@Mocadon We have serious issues in xDrip that need the attention of our lead developer and others.
Can you do the work for this if you feel it is so important and open a PR?

It may seem that everything is working and we have nothing to do.  That is a misconception.    
There are people who cannot use xDrip at all.  We need to address issues that stop people from using xDrip or being able to follow their child etc.

Can you explain how this is a privacy concern? 
If you enable the location service, why would that in itself tell anyone where you are?

**Reply by Mocadon** [reply]
So, I gather it hasn't been done, which is fine, that's one thing I wondered. I'm sure you're busy, and as I wrote, it's not a priority. At some point I'll take a look at the code just cause I'm curious. If I think I can do it maybe I'll pull a copy and give it try. The last time I was a software engineer was 25 years ago though. I wrote some C+ at Schwab, prior to that just some BAL for CICS internals at IBM's research center, and millions of lines of COBOL. Ah, the good old days. 

I didn't mean to offend. And what you folks are doing is wonderful.

**Reply by Navid200** [reply]
There was no offense taken.
If I am overlooking something, which could happen, in prioritizing an issue, it can help if it is clarified.  So, when I asked what the privacy concern was I meant it.

The way I look at this, as long as I don't allow an app to report my location, my location will never be given to anyone whether I enable or disable the service.
But, I would appreciate it if someone disagrees with that expectation to help me see what I have overlooked in that thinking.

**Reply by Mocadon** [reply]
No worries. As I understand it, you're wondering why location is an issue for some people. 'll try to be as clear as I can be. 

So, there are various threat models we look at. For most people any Android or Apple configuration with standard settings and some anti phishing training is enough. Things escalate from there. I am a potential target of State actors, so I'm in the process of locking everything down. The way I'm set up applications are only getting a general location anyway, not a precise one, and not through Google Play Services. Apps may think they are, but they're not.  Also, I'm very confident that the xDrip folks are not a problem in any way.

 In normal Google Android, most location requests are processed by Google Play Services which then stores the location on Google's databases for sale. It does this whether the main location permission is on or not. It can't be prevented. Direct subscribers to this data include every major intelligence agency. Once a device is fingerprinted then deanonymization is fairly straightforward. GrapheneOS prevents this. Out of 100+ apps on this phone, only 2 have location permission and xDrip is one of them. 

So, I hope this explains at least a little bit why some may wonder about the location permission in an application. 

👍

**Reply by Navid200** [reply]
Alright.  Thanks for explaining this.
I will open a submission to add the option.   But, it will be a while before I get to it.  And more importantly, this does not mean that it will be approved.  I cannot say anything about that.  Someone with a lot more experience and knowledge is the one who will be making that call.

Thanks for explaining.

**Reply by jamorham** [reply]
@Mocadon I would say disable location permission for google play services. xDrip only uses actual location for the emergency message feature (if enabled) but android framework requires that location permission is enabled for the app in order to use bluetooth scanning unrelated to any actual use of location service itself at the current target framework level.

**Reply by Mocadon** [reply]
So, in GrapheneOS I've already done that. GrapheneOS will essentially spoof a response by using a fairly inaccurate GPS position. The only way to deny that is at the device level or the app level.

In Google's Android the requesting app would be denied or errored but the request and precise location still gets logged to Google. It's not obvious, but in Google's Android precise location is always provided to Google regardless of any settings if the phone can communicate. 

Again, not a priority in my case since only xDrip is getting the location and Google's getting nothing, but for Google Android users Google is logging a location request from xDrip for profiling. And, since I'm reading it isn't required for Bluetooth since Android 12+, it's something for a less stressful time but a good idea. 

Ah, I remember less stressful times. 👍
